build.gradle

```diff
@@ -403,10 +399,13 @@ project(':iceberg-spark-runtime') {
   }
 
   dependencies {
-    compile project(':iceberg-spark')
+    compileOnly project(':iceberg-spark')
   }
 
   shadowJar {
+    // shade compileOnly dependencies to avoid including in transitive dependencies
+    configurations = [project.configurations.compileOnly]
```

Comment:
I don't think this should exclude all transitive dependencies. How did we test this?

Reply:
`compileOnly` dependencies are not added to artifacts. Using that for the dependencies that we shade makes sense because all of the shaded dependencies should not be listed as dependencies of the shaded Jar.

Reply:
I think the comment is slightly misleading.  Transitive dependencies are *being bundled* in the fat jar, they are just not pulled in when adding the fat jar as an dependency in a gradle or maven project.

Can we reword the comment like so,
Using compileOnly configuration since relying on compile configuration adds transitive artifacts as dependencies of the fat jar

Reply:
Yes, will do.